### ISSUE TITLE
Perhaps a bug in the implementation

### ISSUE THREAD

**Issue by liuquande** [Pulkit-Khandelwal/medical-mldg-seg#1]
https://github.com/Pulkit-Khandelwal/medical-mldg-seg/blob/712c619ad800211ffcb5d148e84a405bdf3c3a48/mldg-seg/Unet3D_meta_learning.py#L115-L161

Hi @Pulkit-Khandelwal , thanks for the code.
Seems the meta loss and step size have been pre-defined in your init function, instead of passing these arguments dynametically in the forward function.
In that case, will the meta-loss be always None? Is it a bug?

**Comment by Pulkit-Khandelwal** [comment]
Hi @liuquande Thanks for the question. See this line here for an example of how the meta_loss and meta_step_size are passed during the training procedure so they get updated: https://github.com/Pulkit-Khandelwal/medical-mldg-seg/blob/712c619ad800211ffcb5d148e84a405bdf3c3a48/mldg-seg/meta_learning_spine.py#L300

Hope this helps!

**Comment by liuquande** [comment]
Hi @Pulkit-Khandelwal ,
Thanks for the reply.

I saw this line. But when passing this arguments into __forward__ function, they are not used anymore. 
You update the value of self.meta_loss and self.meta_step_size, but they are not used.

```
     def forward(self, x, meta_loss, meta_step_size, stop_gradient): 
         self.meta_loss = meta_loss 
         self.meta_step_size = meta_step_size 
         self.stop_gradient = stop_gradient 
  ```

The variable you use in the __init__ function is the pre-defined "meta_loss" and "meta_step_size", instead of the arguments you pass.

Am I right?

**Comment by Pulkit-Khandelwal** [comment]
HI @liuquande, thanks for the comment. Here is the correct file:
https://github.com/Pulkit-Khandelwal/medical-mldg-seg/blob/master/mldg-seg/Unet3D_meta_learning_orig.py

**Comment by liuquande** [comment]
Thanks for the update.
